src/views/viewer/EditMembershipView/EditMembershipView.tsx

```diff
@@ -0,0 +1,192 @@
+      image.onerror = () => resolve(false)
+      image.src = value
+      if (!value) {
+        resolve(true)
+      }
+    })
+
+  const handleUniqueValidation = async (value: string, prevValue?: string) => {
+    if (value === prevValue) {
+      return true
+    }
+    const {
+      data: { membershipByUniqueInput },
+    } = await client.query<GetMembershipQuery, GetMembershipQueryVariables>({
+      query: GetMembershipDocument,
+      variables: { where: { handle: value } },
+    })
+    if (membershipByUniqueInput) {
+      return false
+    } else {
+      return true
+    }
+  }
+
+  const schema = z.object({
+    handle: z
+      .string()
+      .nonempty({ message: 'Member handle cannot be empty' })
+      .min(5, {
+        message: 'Member handle must be at least 5 characters',
+      })
+      .max(40, {
+        message: `Member handle cannot be longer than 40 characters`,
+      })
+      .refine((val) => (val ? MEMBERSHIP_NAME_PATTERN.test(val) : true), {
+        message: 'Member handle may contain only lowercase letters, numbers and underscores',
+      })
+      .refine(
+        async (val) => {
+          const debouncedHandleUniqueValidation = debouncePromise(handleUniqueValidation, 500)
+          const isValid = await debouncedHandleUniqueValidation(val, activeMembership?.handle)
+          return isValid
+        },
+        { message: 'Member handle already in use' }
+      ),
+    avatar: z
+      .string()
+      .max(400)
+      .refine((val) => (val ? URL_PATTERN.test(val) : true), { message: 'Avatar URL must be a valid url' })
+      .refine(
+        async (val) => {
+          const debouncedValidation = debouncePromise(avatarValidation, 500)
+          const isValid = await debouncedValidation(val)
+          return isValid
+        },
+        { message: 'Image not found' }
+      )
+      .optional(),
+    about: z.string().max(1000, { message: 'About cannot be longer than 1000 characters' }).optional(),
+  })
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors, isDirty, isValid },
+  } = useForm<Inputs>({
+    mode: 'onChange',
+    resolver: zodResolver(schema),
+    shouldFocusError: true,
+    defaultValues: {
+      handle: activeMembership?.handle,
+      avatar: activeMembership?.avatarUri,
+      about: activeMembership?.about,
+    },
+  })
+
+  const resetForm = useCallback(() => {
+    reset(
+      {
+        handle: activeMembership?.handle,
+        avatar: activeMembership?.avatarUri,
+        about: activeMembership?.about,
+      },
+      {
+        keepDirty: false,
+      }
+    )
+  }, [activeMembership?.about, activeMembership?.avatarUri, activeMembership?.handle, reset])
+
+  useEffect(() => {
+    if (!activeMembershipLoading && activeMembership) {
+      resetForm()
+    }
+  }, [activeMembership, activeMembershipLoading, resetForm])
+
+  const handleEditMember = handleSubmit(async () => {
+    // TODO
+    // handle submiting edited member
+  })
+
```

Comment:
We dont have the tech for this yet?

Reply:
I looked at: https://github.com/Joystream/membership-faucet/tree/master/src and I don't think we have. 
@kdembler should I do something about it? If yes, I'll probably need some help

Reply:
This will not be faucet's job. Faucet just creates memberships and supplies them with some tokens. This will be a regular extrinsic to update the membership. We can sync in chat about this

Reply:
ok, fixed this. It's possible now